src/main/java/it/gov/pagopa/pu/pagopapayments/connector/pagopa/gpd/GpdServiceImpl.java

```diff
@@ -2,29 +2,111 @@
 
 import it.gov.pagopa.nodo.gpd.dto.generated.PaymentPositionModel;
 import it.gov.pagopa.pu.pagopapayments.connector.pagopa.gpd.client.GpdClient;
+import it.gov.pagopa.pu.pagopapayments.registry.RegistryContextData;
+import it.gov.pagopa.pu.pagopapayments.registry.RegistryEventType;
+import it.gov.pagopa.pu.pagopapayments.registry.RegistryLogger;
+import it.gov.pagopa.pu.pagopapayments.util.Utilities;
+import it.gov.pagopa.pu.registries.dto.generated.RegistryOutcome;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Triple;
 import org.springframework.stereotype.Service;
 
 @Service
+@Slf4j
 public class GpdServiceImpl implements GpdService {
 
   private final GpdClient client;
+  private final RegistryLogger registryLogger;
 
-  public GpdServiceImpl(GpdClient client) {
+  public GpdServiceImpl(GpdClient client, RegistryLogger logger) {
     this.client = client;
+    this.registryLogger = logger;
   }
 
   @Override
   public void paCreatePosition(String apiKey, String organizationfiscalcode, PaymentPositionModel paymentPositionModel) {
-    client.createPosition(apiKey, organizationfiscalcode,paymentPositionModel);
+    registryLogger.execute(
+      getRegistryContextDataFromPaymentPositionModel(
+        organizationfiscalcode,
+        RegistryEventType.createPosition,
+        paymentPositionModel.getIupd(),
+        paymentPositionModel
+      ),
+      paymentPositionModel,
+      () -> {
+        client.createPosition(apiKey, organizationfiscalcode,paymentPositionModel);
+        return Triple.of(null, null, RegistryOutcome.OK);
+      },
+      e -> {
+        log.error(e.getMessage(), e);
+        return null;
+      }
+    );
   }
 
   @Override
   public void paUpdatePosition(String apiKey, String organizationfiscalcode, String iupd, PaymentPositionModel paymentPositionModel) {
-    client.updatePosition(apiKey, organizationfiscalcode,iupd, paymentPositionModel);
+    registryLogger.execute(
+      getRegistryContextDataFromPaymentPositionModel(
+        organizationfiscalcode,
+        RegistryEventType.updatePosition,
+        iupd,
+        paymentPositionModel
+      ),
+      paymentPositionModel,
+      () -> {
+        client.updatePosition(apiKey, organizationfiscalcode, iupd, paymentPositionModel);
+        return Triple.of(null, null, RegistryOutcome.OK);
```

Comment:
let's modify the client class to return the response of the API instead, this way we could provide it to the registryLogger